Support filtering on FILTER field

tomwhite opened the pull request:
Fixes #164

While working on this I found two bugs where the separators between FILTER fields were wrong (VCF and query output).

jeromekelleher commented:
Looks great, thanks @tomwhite. I patched up a gap in the C implementation and added some validation tests. 

The only thing that's missing I think is that we should raise an error on referencing non-existent filters rather than running the set-ops as is. I'm not sure if "." counts here, though - we'd have to try a few bits and bobs against bcftools to see what errors and doesn't.

tomwhite commented:
> Looks great, thanks @tomwhite. I patched up a gap in the C implementation and added some validation tests.

Thanks!

> The only thing that's missing I think is that we should raise an error on referencing non-existent filters rather than running the set-ops as is. I'm not sure if "." counts here, though - we'd have to try a few bits and bobs against bcftools to see what errors and doesn't.

I've handled this case and added a test.

> We should also validate that we get empty sets of variants, which isn't really possible in the current `test_vcf_output` framework. I feel like a specialised function that ran "view" only, and had an expected number of records stated as part of the parametrize parameters would help a lot here.

I'll look at this later.
